app/src/test/java/com/globant/viewmodels/CharacterFragmentViewModelTest.kt

```diff
@@ -0,0 +1,146 @@
+package com.globant.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import com.globant.domain.entities.Comic
+import com.globant.domain.entities.MarvelCharacter
+import com.globant.domain.usecases.GetCharacterByIdUseCase
+import com.globant.domain.usecases.GetComicsUseCase
+import com.globant.domain.utils.Result
+import com.globant.utils.Data
+import com.globant.utils.Status
+import com.google.common.truth.Truth
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.koin.test.AutoCloseKoinTest
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+class CharacterFragmentViewModelTest : AutoCloseKoinTest() {
+
+    companion object {
+        const val VALID_ID = 1017100
+        const val INVALID_ID = -1
+    }
+
+    @ObsoleteCoroutinesApi
+    var mainThreadSurrogate = newSingleThreadContext("UI thread")
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    lateinit var viewModel: CharacterFragmentViewModel
+    @Mock
+    lateinit var marvelCharacterSuccess: Result.Success<MarvelCharacter>
+    @Mock
+    lateinit var marvelCharacterFailure: Result.Failure
+    @Mock
+    lateinit var marvelCharacter: MarvelCharacter
+    @Mock
+    lateinit var comicSuccess: Result.Success<List<Comic>>
+    @Mock
+    lateinit var comicFailure: Result.Failure
+    @Mock
+    lateinit var comics: List<Comic>
+    @Mock
+    lateinit var exception: Exception
+    @Mock
+    lateinit var getCharacterByIdUseCase: GetCharacterByIdUseCase
+    @Mock
+    lateinit var getComicsUseCase: GetComicsUseCase
+
+    @ExperimentalCoroutinesApi
+    @ObsoleteCoroutinesApi
+    @Before
+    fun setup() {
+        Dispatchers.setMain(mainThreadSurrogate)
+        MockitoAnnotations.initMocks(this)
+        viewModel = CharacterFragmentViewModel(getCharacterByIdUseCase, getComicsUseCase)
+    }
+
+    @ExperimentalCoroutinesApi
+    @ObsoleteCoroutinesApi
+    @After
+    fun after() {
+        mainThreadSurrogate.close()
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun getCharacterByIdSuccess() {
+        val liveDataUnderTest = viewModel.mainState.testObserver()
+        Mockito.`when`(getCharacterByIdUseCase.invoke(VALID_ID, true)).thenReturn(marvelCharacterSuccess)
```

Comment:
use whenever from MockitoKotlin2 instead of this when pls

Reply:
done